### PR TITLE
use GetCapLast() to get CAP_LAST_CAP and avoid to read file "/proc/sys/kernel/cap_last_cap" everytime when importing

### LINKED ISSUE #26: import  github.com/syndtr/gocapability will auto read /proc/sys/kernel/cap_last_cap
https://github.com/syndtr/gocapability/blob/42c35b4376354fd554efc7ad35e0b7f94e3a0ffb/capability/capability_linux.go#L47

runc  --help will read file  "/proc/sys/kernel/cap_last_cap" I want read it when I need.

### PR DESCRIPTION
**Pull request by ningmingxiao**
fix https://github.com/syndtr/gocapability/issues/26

### CODE REVIEW

**Comment by avagin** [comment]
pls update the commit message to explain why the change is needed.

File: capability/capability_linux.go [hunk 1]
```diff
@@ -27,23 +28,31 @@ const (
...
-			capUpperMask = 0
+
+func GetCapMaskCapLast() (uint32,Cap) {
+	once.Do(func() {
+		var hdr capHeader
+		capget(&hdr, nil)
+		capVers = hdr.version
+	
+		if initLastCap() == nil {
+			CAP_LAST_CAP = capLastCap
```
Comment (avagin) [inline capability/capability_linux.go:41]: CAP_LAST_CAP is public.The external code can use it and after this change it will be broken.

Comment (ningmingxiao) [reply capability/capability_linux.go:41]: It is not compatible with old versions after my pr, do you have some idea?

Comment (ningmingxiao) [reply capability/capability_linux.go:41]: I try to solve this problem to make runc run faster .
https://github.com/opencontainers/runc/issues/4356

File: capability/capability_linux.go [hunk 2]
```diff
@@ -27,23 +28,31 @@ const (
...
-	capVers = hdr.version
 
-	if initLastCap() == nil {
-		CAP_LAST_CAP = capLastCap
-		if capLastCap > 31 {
-			capUpperMask = (uint32(1) << (uint(capLastCap) - 31)) - 1
-		} else {
-			capUpperMask = 0
+
+func GetCapMaskCapLast() (uint32,Cap) {
```
Comment (avagin) [inline capability/capability_linux.go]: If you want to add a public function, you need to add a documentation for it.

File: capability/capability_linux.go [hunk 3]
```diff
@@ -27,23 +28,31 @@ const (
...
+			CAP_LAST_CAP = capLastCap
+			if capLastCap > 31 {
+				capUpperMask = (uint32(1) << (uint(capLastCap) - 31)) - 1
+			} else {
+				capUpperMask = 0
+			}
 		}
-	}
+	})
+	return capUpperMask,CAP_LAST_CAP
```
Comment (avagin) [inline capability/capability_linux.go]: capUpperMask doesn't look like something that should be shared externally.

File: capability/capability_linux.go [hunk 4]
```diff
@@ -27,21 +28,32 @@ const (
...
-		} else {
-			capUpperMask = 0
+func getCapMaskCapLast() (uint32, Cap) {
+	once.Do(func() {
+		var hdr capHeader
+		capget(&hdr, nil)
+		capVers = hdr.version
+
+		if initLastCap() == nil {
+			CAP_LAST_CAP = capLastCap
```
Comment (avagin) [inline capability/capability_linux.go:41]: I think you can remove CAP_LAST_CAP to be sure that old users notice the change and update their code.

BTW: @kolyshkin created the fork https://github.com/kolyshkin/capability and he is going to support it.

**Pull request closed without merging**